docsk8s/troubleshooting/ls-k8s-troubleshooting-methods.asciidoc

```diff
@@ -92,4 +94,30 @@ jdk/bin/jcmd 1 GC.heap_dump /tmp/heap_dump.hprof
 [source,bash]
 --
 kubectl cp logstash-7477d46bb7-4lcnv:/tmp/heap_dump.hprof ./heap.hprof
---
\ No newline at end of file
+--
+
+[[ls-k8s-pq-util]]
+=== Running PQ utility
+
+In case the pod is terminated ungracefully and hence the persistent queues are corrupted, `pqcheck` and `pqrepair` are the tools for troubleshooting.
+
+Run {logstash-ref}/persistent-queues.html#pqcheck[pqcheck] to identify corrupted files: 
+
+[source,bash]
+--
+kubectl exec logstash-0 -it -- /usr/share/logstash/bin/pqcheck /usr/share/logstash/data/queue/pipeline_id
+--
+
+Run {logstash-ref}/persistent-queues.html#pqrepair[pqrepair] to repair the queue: 
+
+[source,bash]
+--
+kubectl exec logstash-0 -it -- /usr/share/logstash/bin/pqrepair /usr/share/logstash/data/queue/pipeline_id
+--
+
+[[ls-k8s-pq-drain]]
+=== Draining the PQ
+
+Set `queue.drain: true` to drain and wait for the persistent queues to empty when receive SIGTERM. However, the default `TerminationGracePeriodSeconds` in {k8s} is 30 seconds. If {ls} is still running after the grace period has elapsed, {k8s} sends a SIGKILL to terminate the pod immediately. Hence, {ls} cannot finish draining.
+
+A workaround is to set a very long period like 1 year to `TerminationGracePeriodSeconds` to make sure that {ls} has enough time to drain all events.
```

Comment:
```suggestion
To avoid {ls} shutting down before the queue is completely drained, we recommend setting the `TerminationGracePeriodSeconds` value to an artifically long period, such as 1 year, to give {ls} sufficient time to drain the queue when this functionality is required. 
```